### PR TITLE
Deal with friends from multiple classes

### PR DESCRIPTION
**Pull request by LarryOsterman**

### CODE REVIEW

File: tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp [hunk 1]
```diff
@@ -1318,6 +1318,45 @@ static std::string GetNavigationId(clang::FunctionDecl const* func)
   return navigationId;
 }
 
+static std::string GetNavigationId(
+    clang::FriendDecl const* friendDecl,
+    std::string const& parentNodeNavigationId)
+{
+  std::string navigationId = parentNodeNavigationId + "_friend_";
```
Comment (RickWinter) [inline tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp:1325]: Can the parentNodeNavigationId be ""? Can we assert these premises.

File: tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp [hunk 2]
```diff
@@ -1318,6 +1318,45 @@ static std::string GetNavigationId(clang::FunctionDecl const* func)
   return navigationId;
 }
 
+static std::string GetNavigationId(
```
Comment (RickWinter) [inline tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp:1321]: Add function comment

File: tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp [hunk 3]
```diff
@@ -1318,6 +1318,45 @@ static std::string GetNavigationId(clang::FunctionDecl const* func)
...
+    {
+      navigationId
+          += QualType::getAsString(friendDecl->getFriendType()->getType().split(), LangOptions{});
+    }
+    else
+    {
+      friendDecl->dump(llvm::outs());
+    }
+  }
+  return navigationId;
```
Comment (RickWinter) [inline tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp:1357]: If you end up in the final else of the initial if/else then you could end up with navigationId being `<parentNodeId>_friend_`. Is a navigationId ending in "_" a valid format?

**Review by RickWinter — APPROVED** [review]

File: tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp (hunk 1 above)
Comment (LarryOsterman) [reply tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp:1325]: It cannot be empty, I'll add an error message if it is.

File: tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp (hunk 3 above)
Comment (LarryOsterman) [reply tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp:1357]: All strings are valid, the only restriction is that the names must be unique.

**Pull request merged**